src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```diff
@@ -32,6 +33,8 @@ public class ParameterProvider {
   public static final int IO_TIME_CPU_RATIO_DEFAULT = 2;
   public static final int BLOB_UPLOAD_MAX_RETRY_COUNT_DEFAULT = 24;
   public static final long MAX_MEMORY_LIMIT_IN_BYTES_DEFAULT = -1L;
+  /* parameter that enables main memory optimization for Parquet format (BdecVersion.THREE) */
+  public static final boolean ENABLE_PARQUET_MEM_OPT_DEFAULT = false;
```

Comment:
I was wondering whether we should rather call this parameter everywhere: `ENABLE_PARQUET_INTERNAL_BUFFERING / enableParquetInternalBuffers` and include in comments what it is needed for (mem opt) and why it leads to mem opt (less mem comparing to java objs).

Reply:
done, renamed and extended comment.